### PR TITLE
Prevent duplication of dom elements with similar ids

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

Makes a correction in the commercial code to prevent a problem that we noticed in DCR. This will not affect frontend as in essence we are just adding a precautionary step. 

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No

### CODE REVIEW

**Review by ioanna0 — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/create-slots.js [hunk 1]
```diff
@@ -143,16 +143,30 @@ const adSlotDefinitions = {
...
+
+    // The next three lines are to prevent a problem that appeared with DCR
+    // by which an AdSlot with the given id might have already been
+    // statically introduced server side and needs to be removed before
+    // this code takes effect on the DOM.
+    // We are simply making sure that if we are about to introduce a DOM
+    // element with a given id, that we ensure there isn't already one.
+    const element = document.getElementById(id);
+    if (element && element.parentNode) {
+        element.parentNode.removeChild(element);
```
Comment (janua) [inline static/src/javascripts/projects/commercial/modules/dfp/create-slots.js]: Does this need to happen in fastdom?

Comment (jeteve) [reply static/src/javascripts/projects/commercial/modules/dfp/create-slots.js]: Maybe console.log an error there? It's good to defend against the duplicated ID situation, but if the code was sane, il shouldn't happen. A fat error in the log would prompt us to fix any misbehaving code.

Comment (shtukas) [reply static/src/javascripts/projects/commercial/modules/dfp/create-slots.js]: @jeteve Happy to add a `console.log`m but technically this is not sign of a problem per se, just frontend and dotcom-rendering having different logic :)

**Comment by shtukas** [comment]
@jeteve @janua In an additional commit I added the console log and limited the effect to DCR

File: static/src/javascripts/projects/commercial/modules/dfp/create-slots.js (hunk 1 above)
Comment (janua) [reply static/src/javascripts/projects/commercial/modules/dfp/create-slots.js]: @shtukas What happened to my query about fastdom? It seems my comment was just ignored.

Comment (shtukas) [reply static/src/javascripts/projects/commercial/modules/dfp/create-slots.js]: Discussed IRL :)

**Comment by shtukas** [comment]
Follow up: https://github.com/guardian/frontend/pull/22330

**Pull request merged**